kubernetes/client.go

```diff
@@ -99,30 +99,24 @@ func NewClient() (*IstioClient, error) {
 
 // It returns a list of all namespaces/projects of the cluster.
 // It returns an error on any problem.
-func (in *IstioClient) GetNamespaces() ([]string, error) {
+func (in *IstioClient) GetNamespaces() (*v1.NamespaceList, error) {
 	namespaces, err := in.k8s.CoreV1().Namespaces().List(emptyListOptions)
 	if err != nil {
 		return nil, err
 	}
-	names := make([]string, len(namespaces.Items))
-	for i, namespace := range namespaces.Items {
-		names[i] = namespace.Name
-	}
-	return names, nil
+
+	return namespaces, nil
 }
 
 // It returns a list of services for a given namespace.
 // It returns an error on any problem.
-func (in *IstioClient) GetServices(namespace string) ([]string, error) {
+func (in *IstioClient) GetServices(namespace string) (*v1.ServiceList, error) {
```

Comment:
it will be better to be more clear with the param name what is it representing in namespace. name? id? any other field?

Reply:
you are right! It improve readability :+1: